fix(types): move @oxc-project/types to dependencies

ottomated opened the pull request:
Hotfix. NPM doesn't install devDependencies automatically, so types were missing.

ottomated commented:
* **#6909** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6909?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ottomated and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Oct 25, 10:25 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

ottomated committed:
fix(types): move @oxc-project/types to dependencies (#6909)

Hotfix. NPM doesn't install devDependencies automatically, so types were missing.
